src/createStore.js

```diff
@@ -93,7 +94,17 @@ export default function createStore(reducer, initialState) {
       'Actions must be plain objects. Use custom middleware for async actions.'
     );
 
-    currentState = currentReducer(currentState, action);
+    if (isDispatching === true) {
+      throw new Error('Reducers may not dispatch actions.');
+    }
+
+    try {
```

Comment:
We can't add a try {} catch {} here. This will cause way more problems than it solves. All reducer errors will be swallowed.